pages/bridge.tsx

```diff
@@ -23,13 +32,92 @@ const Bridge = () => {
   const [bridgeTarget, setBridgeTarget] = useState<BridgeTarget>(
     BridgeTarget.L2
   );
+  const { address } = useAccount();
+  const [amount, setAmount] = useState<string>("0");
+
+  const rawAmount = parseUnits(amount, l1.token?.decimals || 18);
+
+  // 1️⃣ l1 bridge to l2
+  const { data: allowanceL1 } = useContractRead({
+    account: address,
+    chainId: l1Config.chain.id,
+    address: l1Config.tokenAddress,
+    abi: parseAbi([
+      "function allowance(address, address) public view returns (uint256)",
```

Comment:
It might be good to  move these abis to a constants file in case we ever need to make changes